app/src/app/components/contents/common/DataTable.tsx

```diff
@@ -0,0 +1,47 @@
+import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from "@tanstack/react-table";
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "../../Base/Table";
+
+interface DataTableProps<TData> {
+  columns: ColumnDef<TData, any>[];
+  data: TData[];
+}
+
+export const DataTable = <TData,>({ data, columns }: DataTableProps<TData>) => {
+  const table = useReactTable({ columns, data, getCoreRowModel: getCoreRowModel() });
+  const rows = table.getRowModel().rows;
+
+  return (
+    <div className="rounded-md border">
+      <Table>
+        <TableHeader>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <TableRow key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <TableHead key={header.id}>
+                  {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
```

Comment:
What does `isPlaceholder` mean here? Why would a column be a placeholder.

Reply:
its a tanstack table thing where there can be a placeholer header